src/Stateless/Reflection/MethodInfo.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Stateless.Reflection
+{
+    /// <summary>
+    /// Describes an internal method through the reflection API.
+    /// </summary>
+    public class MethodInfo
```

Comment:
Sounds good 👍  though it'd be nice for it to appear only under the _Reflection_ part of the public API (so we'd still need to rename it `InvocationInfo` or similar).